cmd/maya-apiserver/webhook/start.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/pkg/webhook"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+)
+
+var (
+	masterURL  string
+	kubeconfig string
+)
+
+func main() {
+	var parameters webhook.WParameters
+
+	// get command line parameters
+	flag.IntVar(&parameters.Port, "port", 443, "Webhook server port.")
+	flag.StringVar(&parameters.CertFile, "tlsCertFile", "/etc/webhook/certs/cert.pem", "File containing the x509 Certificate for HTTPS.")
+	flag.StringVar(&parameters.KeyFile, "tlsKeyFile", "/etc/webhook/certs/key.pem", "File containing the x509 private key to --tlsCertFile.")
+	flag.Parse()
+
+	// Get in cluster config
+	cfg, err := getClusterConfig(kubeconfig)
+	if err != nil {
+		glog.Errorf("Error building kubeconfig: %s", err.Error())
+	}
+	// Building Kubernetes Clientset
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		glog.Errorf("Error building kubernetes clientset: %s", err.Error())
+	}
+	// Building OpenEBS Clientset
+	openebsClient, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		glog.Errorf("Error building openebs clientset: %s", err.Error())
+	}
+
+	wh, err := webhook.NewWebhook(parameters, kubeClient, openebsClient)
+	if err != nil {
+		glog.Errorf("failed to create validation webhook: %s", err.Error())
+	}
+
+	// define http server and server handler
+	mux := http.NewServeMux()
+	mux.HandleFunc("/validate", wh.Serve)
+	wh.Server.Handler = mux
+
+	// start webhook server in new routine
+	go func() {
+		if err := wh.Server.ListenAndServeTLS("", ""); err != nil {
+			glog.Errorf("Failed to listen and serve webhook server: %v", err)
+		}
+	}()
+
+	glog.Info("Webhook server started")
+
+	// listening OS shutdown singal
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
+	<-signalChan
+
+	glog.Infof("Got OS shutdown signal, shutting down webhook server gracefully...")
+	wh.Server.Shutdown(context.Background())
+}
+
+// GetClusterConfig return the config for k8s.
```

Comment:
n_p: In comment ( `GetClusterConfig` -->`getClusterConfig`)

Reply:
thanks.. will fix the nit pickings